### ISSUE TITLE
If you log into /app, then /develop, then /app again, and then start doing E2E... folks cannot send E2E to you due to bad OTK MACs

### ISSUE THREAD

**Issue by ara4n** [vector-im/riot-web#6196]
The error looks like: https://github.com/vector-im/riot-web/issues/6195

Rageshake: https://github.com/matrix-org/riot-web-rageshakes/issues/687

**Comment by ara4n** [comment]
@richvdh says this is because when on /develop the app will have published OTKs whose private half then get lost in indexeddb when you move back to /app.

However, I'm confused on why the app would ever be publishing OTKs on /develop, if it's not participating in E2E rooms?

**Comment by richvdh** [comment]
If you're not participating in e2e rooms, how do you know that people can't send you e2e messages? In general the app publishes new otks whenever it decides the number on the server is getting a bit low (it aims to keep 50 on the server, IIRC).

Actually the problem is a bit deeper than OTKs, though. When you switch to develop, all your e2e state is migrated to indexeddb. When you then switch back to app, it sees that you don't have any e2e state and sets you up a whole new OlmAccount, with new signing keys. Which it then uses to sign any new OTKs, which leads to the "Unable to verify signature" error from any other clients which saw your original keys.

Basically by switching back and forth you have completely split-brained your e2e state, and tbh all bets are off as to what will work in the future.

The app does try to warn you of this when the migration happens: https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/components/structures/MatrixChat.js#L1178.

Short of an interim release that supports indexeddb but doesn't actually do the migration, I'm not sure how we could do better here.

**Comment by ara4n** [comment]
> If you're not participating in e2e rooms, how do you know that people can't send you e2e messages?

The bug happens when you start trying to e2e for the first time (after a app->develop->app shuffle).

Eitherway, if we end up creating a whole new OlmAccount on the develop->app transition, that explains why the existing published OTKs are completely out of sync and everything breaks.  Nothing we can do here, short of release a riot-web to get everyone on develop asap.
